Managers/PerformanceMan.cpp

```diff
@@ -0,0 +1,264 @@
+
+			sprintf_s(str, sizeof(str), "Objects: %i", g_MovableMan.GetKnownObjectsCount());
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX, c_StatsHeight + 60, str, GUIFont::Left);
+
+			sprintf_s(str, sizeof(str), "MOIDs: %i", g_MovableMan.GetMOIDCount());
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX, c_StatsHeight + 70, str, GUIFont::Left);
+
+			sprintf_s(str, sizeof(str), "Sim Updates Since Last Drawn: %i", g_TimerMan.SimUpdatesSinceDrawn());
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX, c_StatsHeight + 80, str, GUIFont::Left);
+
+			if (g_TimerMan.IsOneSimUpdatePerFrame()) { g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX, c_StatsHeight + 90, "ONE Sim Update Per Frame!", GUIFont::Left); }
+
+			sprintf_s(str, sizeof(str), "Sound channels: %d / %d ", g_AudioMan.GetPlayingChannelCount(), g_AudioMan.GetTotalChannelCount());
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX, c_StatsHeight + 100, str, GUIFont::Left);
+
+			if (m_AdvancedPerfStats) { DrawPeformanceGraphs(bitmapToDrawTo); }
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PerformanceMan::DrawPeformanceGraphs(AllegroBitmap bitmapToDrawTo) {
+		// Update current sample percentage
+		CalculateSamplePercentages();
+
+		char str[512];
+
+		//Draw advanced performance counters
+		for (unsigned short pc = 0; pc < PERF_COUNT; ++pc) {
+			unsigned short blockStart = c_GraphsStartOffsetY + pc * c_GraphBlockHeight;
+
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX, blockStart, m_PerfCounterNames[pc], GUIFont::Left);
+
+			// Print percentage from PerformanceCounters::PERF_SIM_TOTAL
+			unsigned short perc = static_cast<unsigned short>((static_cast<float>(GetPerormanceCounterAverage(static_cast<PerformanceCounters>(pc))) / static_cast<float>(GetPerormanceCounterAverage(PERF_SIM_TOTAL)) * 100));
+			sprintf_s(str, sizeof(str), "%%: %u", perc);
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX + 60, blockStart, str, GUIFont::Left);
+
+			// Print average processing time in ms
+			sprintf_s(str, sizeof(str), "T: %lli", GetPerormanceCounterAverage(static_cast<PerformanceCounters>(pc)) / 1000);
+			g_FrameMan.GetLargeFont()->DrawAligned(&bitmapToDrawTo, c_StatsOffsetX + 96, blockStart, str, GUIFont::Left);
+
+			unsigned short graphStart = blockStart + c_GraphsOffsetX;
+
+			//Draw graph backgrounds
+			bitmapToDrawTo.DrawRectangle(c_StatsOffsetX, graphStart, c_MaxSamples, c_GraphHeight, 240, true);
+			bitmapToDrawTo.DrawLine(c_StatsOffsetX, graphStart + c_GraphHeight / 2, c_StatsOffsetX + c_MaxSamples, graphStart + c_GraphHeight / 2, 96);
+
+			//Reset peak value
+			unsigned short peak = 0;
+
+			//Draw sample dots
+			unsigned short smpl = m_Sample;
```

Comment:
This should probably be pulled into a private method so it's at least out of sight. Make it return peak I guess, so you can use that where you need to.

Reply:
It's not a big chunk of code and very much part of the graphs.
So I don't really see a reason to do that because it won't make much of a difference.